pkg/api/api.go

```diff
@@ -24,6 +24,13 @@ type BulkResponse struct {
 	Items  []map[string]any `json:"items,omitempty"`
 }
 
+type dataStore struct {
+	// three dimensional map storing index, id and the document
+	data map[string]map[string]map[string]any
+}
+
+var doc dataStore = dataStore{data: make(map[string]map[string]map[string]any)}
```

Comment:
Reads and writes to this variable can occur in parallel by multiple requests, so you would need a mutex to serialize access.